homeassistant/components/amberelectric/config_flow.py

```diff
@@ -3,18 +3,47 @@
 
 import amberelectric
 from amberelectric.api import amber_api
-from amberelectric.model.site import Site
+from amberelectric.model.site import Site, SiteStatus
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_API_TOKEN
 from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
 
-from .const import CONF_SITE_ID, CONF_SITE_NAME, CONF_SITE_NMI, DOMAIN
+from .const import CONF_SITE_ID, CONF_SITE_NAME, DOMAIN
 
 API_URL = "https://app.amber.com.au/developers"
 
 
+def generate_site_selector_name(site: Site) -> str:
+    """Generate the name to show in the site drop down in the configuration flow."""
+    if site.status == SiteStatus.CLOSED:
+        return site.nmi + " (Closed: " + site.closed_on.isoformat() + ")"  # type: ignore[no-any-return]
+    if site.status == SiteStatus.PENDING:
+        return site.nmi + " (Pending)"  # type: ignore[no-any-return]
+    return site.nmi  # type: ignore[no-any-return]
+
+
+def filter_sites(sites: list[Site]) -> list[Site]:
+    """Deduplicates the list of sites."""
+    filtered: list[Site] = []
+
+    for site in sorted(sites, key=lambda site: site.status.value):
+        if (
+            site.status == SiteStatus.ACTIVE
+            or len([s for s in filtered if s.nmi == site.nmi]) == 0
```

Comment:
```suggestion
            or site.nmi not in filtered_nmi
```

So we avoid looping over all filtered sites